client/my-sites/themes/install-theme-button.tsx

```diff
@@ -1,18 +1,27 @@
 import { FEATURE_INSTALL_THEMES } from '@automattic/calypso-products';
 import { Button } from '@automattic/components';
 import { translate } from 'i18n-calypso';
-import { connect } from 'react-redux';
+import { connect, ConnectedProps } from 'react-redux';
+import { Dispatch } from 'redux';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { isJetpackSiteMultiSite, isJetpackSite } from 'calypso/state/sites/selectors';
+import { IAppState } from 'calypso/state/types';
 import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 import { trackClick } from './helpers';
-
 import './install-theme-button.scss';
 
-const InstallThemeButton = ( {
+interface TracksEventProps {
+	tracksEventProps: { site_type: string | null };
+}
+
+interface InstallThemeButtonProps extends PropsFromRedux {
+	canUploadThemesOrPlugins: boolean;
```

Comment:
Making this prop as required will break existing consumer components like `ThemeHeader`.